src/archive/src/main.rs

```diff
@@ -421,54 +422,49 @@ fn http_request(req: HttpRequest) -> HttpResponse {
 fn encode_metrics(w: &mut MetricsEncoder<Vec<u8>>) -> std::io::Result<()> {
     with_config(|config| {
         w.encode_gauge(
-            "ii_archive_last_upgrade_timestamp",
-            config.last_upgrade_timestamp as f64,
+            "ii_archive_last_upgrade_timestamp_seconds",
+            Duration::from_nanos(config.last_upgrade_timestamp).as_secs_f64(),
             "Timestamp of the last upgrade of this canister.",
         )
     })?;
     with_log(|log| {
         w.encode_gauge(
-            "ii_archive_log_entries_count",
+            "ii_archive_entries_count{source=\"log\"}",
             log.len() as f64,
             "Number of log entries stored in this canister.",
         )?;
```

Comment:
The latest version of `metrics_encoder` has [explicit support for labeled metrics](https://gitlab.com/dfinity-lab/public/ic/-/blob/86d4fe8bc8e62a31776a42de0aebb26a745ce32a/rs/monitoring/metrics_encoder/src/lib.rs#L161-178) and [checks against non-alphanumeric characters in metric names](https://gitlab.com/dfinity-lab/public/ic/-/blob/86d4fe8bc8e62a31776a42de0aebb26a745ce32a/rs/monitoring/metrics_encoder/src/lib.rs#L229-246), meaning this change would start failing as soon as you update the `metrics_encoder` dependency, even though it will work for now.

I would pull in the latest version of `metrics_encoder` and use `w.gauge_vec` here instead.

Reply:
Thanks!